src/global_distrib/mod_global_distrib_utils.erl

```diff
@@ -155,18 +157,22 @@ create_ets(Name, Type) ->
 -spec resolve_endpoints([{inet:ip_address() | string(), inet:port_number()}]) ->
                                [endpoint()].
 resolve_endpoints(Endpoints) ->
-    lists:flatmap(
-      fun({Addr, Port}) ->
-              case to_ip_tuples(Addr) of
-                  {ok, IpAddrs} ->
-                      [{IpAddr, Port} || IpAddr <- IpAddrs];
-                  {error, {Reasonv6, Reasonv4}} ->
-                      ?ERROR_MSG("Cannot convert ~p to IP address: IPv6: ~s. IPv4: ~s.",
-                                 [Addr, inet:format_error(Reasonv6), inet:format_error(Reasonv4)]),
-                      error({Reasonv6, Reasonv4})
-              end
-      end,
-      Endpoints).
+    lists:foldl(fun resolve_endpoint/2, [], Endpoints).
+
+resolve_endpoint({Addr, _Port} = E, Endpoints) when is_tuple(Addr) ->
+    [E | Endpoints];
+resolve_endpoint({Addr, Port}, Endpoints) ->
+    case to_ip_tuples(Addr) of
+        {ok, IpAddrs} ->
+            Resolved = Endpoints ++ [{IpAddr, Port} || IpAddr <- IpAddrs],
```

Comment:
This is pretty much precisely why I `flatmap`ed it before. I think it would do well for this code too to replace `lists:foldl` with `lists:flatmap` in `resolve_endpoints/1`